incubator/hnc/internal/validators/object.go

```diff
@@ -77,6 +71,12 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, inst *unstructured
 		return allow("")
 	}
 
+	// Allow changes to the types that are not in propagate mode.
```

Comment:
We can use `request.Kind` [link](https://godoc.org/k8s.io/api/admission/v1beta1#AdmissionRequest) and have this check much earlier in the flow, even before calling Decode(). In fact, I'd make this the very first check we do, even before calling `isHNCServiceAccount()`.

Reply:
sg. Will move it to the top.